packages/protocol/contracts/layer2/core/Anchor.sol

```diff
@@ -222,10 +221,17 @@ contract Anchor is Ownable2Step, ReentrancyGuard {
         onlyValidSender
         nonReentrant
     {
-        if (_blockParams.blockIndex == 0) {
-            _validateProposal(_proposalParams);
+        uint48 lastProposalId = _proposalState.proposalId;
+
+        if (_proposalParams.proposalId < lastProposalId) {
```

Comment:
```suggestion
        if (
            _proposalParams.proposalId != lastProposalId &&
            _proposalParams.proposalId != lastProposalId + 1
        ) {
```
We have only two valid states for `proposalId`: the current one or increased by one. All other changes are not possible to include on L1 since
https://github.com/taikoxyz/taiko-mono/blob/a223e2285c69084e85568246913ebe0322a3a1bb/packages/protocol/contracts/layer1/core/impl/Inbox.sol#L252

Reply:
Really???